test/cases/ostree-ami-image.sh

```diff
@@ -0,0 +1,1105 @@
+aws s3 cp "${IGNITION_CONFIG_SAMPLE_PATH}" "${BUCKET_URL}/" --acl public-read
+
+############################################################
+##
+## Build edge-ami
+##
+############################################################
+
+# Write a blueprint for raw ami.
+tee "$BLUEPRINT_FILE" > /dev/null << EOF
+name = "ami"
+description = "A rhel-edge ami"
+version = "0.0.1"
+modules = []
+groups = []
+
+[[customizations.user]]
+name = "admin"
+description = "Administrator account"
+password = "\$6\$GRmb7S0p8vsYmXzH\$o0E020S.9JQGaHkszoog4ha4AQVs3sk8q0DvLjSMxoxHBKnB2FBXGQ/OkwZQfW/76ktHd0NX5nls2LPxPuUdl."
+key = "${SSH_KEY_PUB}"
+home = "/home/admin/"
+groups = ["wheel"]
+
+[customizations.ignition.firstboot]
+url = "${OBJECT_URL}/config.ign"
+EOF
+
+greenprint "📄 aws ami blueprint"
+cat "$BLUEPRINT_FILE"
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing edge ami blueprint"
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+sudo composer-cli blueprints depsolve ami
+
+# Build ami.
+build_image ami "${AMI_IMAGE_TYPE}" "${PROD_REPO_URL}"
+
+# Download the image
+greenprint "📥 Downloading the ami image"
+sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
+AMI_FILENAME="${COMPOSE_ID}-${AMI_IMAGE_FILENAME}"
+# Configure ami file with correct permissions
+sudo chmod +r "${AMI_FILENAME}"
+
+# Upload ami to AWS S3 bucket
+greenprint "📂 Upload raw ami to S3 bucket"
+aws s3 cp "${AMI_FILENAME}" "${BUCKET_URL}/" --acl public-read
+
+greenprint "🧹 Clean up edge-ami compose and blueprint"
+sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
+sudo composer-cli blueprints delete ami > /dev/null
+
+# Create container simple file
+CONTAINERS_FILE=containers.json
+
+tee "$CONTAINERS_FILE" > /dev/null << EOF
+{
+  "Description": "${AMI_FILENAME}",
+  "Format": "raw",
+  "Url": "${BUCKET_URL}/${AMI_FILENAME}"
+}
+EOF
+
+# Import the image as an EBS snapshot into EC2
+IMPORT_SNAPSHOT_INFO=output_snapshot_info.json
+aws ec2 import-snapshot --description "RHEL edge ami snapshot" --disk-container file://"${CONTAINERS_FILE}" > "${IMPORT_SNAPSHOT_INFO}"
+IMPORT_TASK_ID=$(cat "${IMPORT_SNAPSHOT_INFO}" | jq -r '.ImportTaskId')
+
+# Monitor snapshot status
+greenprint "Check import status of the snapshot"
+IMPORT_SNAPSHOT_TASK=output_snapshot_task.json
+while true; do
+    aws ec2 describe-import-snapshot-tasks --import-task-ids "${IMPORT_TASK_ID}" | tee "${IMPORT_SNAPSHOT_TASK}" > /dev/null
+    IMPORT_STATUS=$(cat "${IMPORT_SNAPSHOT_TASK}" | jq -r '.ImportSnapshotTasks[].SnapshotTaskDetail.Status')
+
+    # Have the snapshot finished?
+    if [[ $IMPORT_STATUS != active ]]; then
+        break
+    fi
+    
+    # Wait 5 seconds and try again.
+    sleep 5
+done
+
+if [[ $IMPORT_STATUS != completed ]]; then
+  echo "Something went wrong with the snapshot. 😢"
+  exit 1
+else
+  greenprint "Snapshot imported successfully."
+fi
+
+
+# Import  keypair
+greenprint "Share ssh public key with AWS"
```

Comment:
All the following things, like keypair, vpc, internet gateway, subnet, you have two choices to configure them for test running in parallel support.

1. each test uses its own above things, their name should be random
2. each test uses the same above things, you need check them existing first. If they exist, just ignore configuration.